community/kernel/src/main/java/org/neo4j/helpers/Settings.java

```diff
@@ -47,10 +47,13 @@
  * which means that you don't want any default value at all, and MANDATORY, which means that the user has to specify a value
  * for this setting. Not providing a mandatory value for a setting leads to an IllegalArgumentException.
  *
- * If a setting does not have a provided value, and no default, then
+ * @deprecated this class is deprecated and will be moved to internal packages in the next major release
  */
+@Deprecated
```

Comment:
+1, we can't remove this in 3.0 unless it is deprecated in a 2.x release
On Dec 8, 2015 9:42 AM, "Davide Grohmann" <notifications@github.com> wrote:

> In community/kernel/src/main/java/org/neo4j/helpers/Settings.java
> <https://github.com/neo4j/neo4j/pull/6029#discussion_r46932253>:
>
> >   */
> > +@Deprecated
>
> I guess because this PR is targeting 2.3
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/neo4j/neo4j/pull/6029/files#r46932253>.
>